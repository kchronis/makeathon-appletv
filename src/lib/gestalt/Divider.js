/**
 * WIP - Gestalt like Divider
 * https://github.com/pinterest/gestalt/tree/master/src/Button
 * @flow
 */

import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import type { StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import PIColor from '../styles/picolor';

type Props = {
  style?: StyleObj, // React Native only: Should go away eventually
};

export default function Divider(props: Props) {
  return <View style={[styles.divider, props.style]} />;
}

const styles = StyleSheet.create({
  divider: {
    backgroundColor: PIColor.pinterestSuperLightGrayColor,
    height: StyleSheet.hairlineWidth,
  },
});
