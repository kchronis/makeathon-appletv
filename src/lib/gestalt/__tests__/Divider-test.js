/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import Divider from '../Divider';

test('<Divider />', () => {
  const tree = renderer.create(<Divider />).toJSON();
  expect(tree).toMatchSnapshot();
});
