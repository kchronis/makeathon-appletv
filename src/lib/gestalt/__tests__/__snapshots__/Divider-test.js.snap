// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EFEFEF",
        "height": 0.5,
      },
      undefined,
    ]
  }
/>
`;
